src/DynamicNLPModels.jl

```diff
@@ -366,14 +367,28 @@ function _build_sparse_lq_dynamic_model(dnlp::LQDynamicData{T, V, M, MK}) where
     gl = dnlp.gl
     gu = dnlp.gu
 
-    H   = _build_H(Q, R, N; Qf = Qf, S = S)
-    J1  = _build_sparse_J1(A, B, N)
-    J2  = _build_sparse_J2(E, F, N)
+    nc = size(E, 1)
 
-    J = vcat(J1, J2)
+    H_colptr = _init_similar(Int[], ns * (N + 1) + nu * N + 1, Int)
```

Comment:
until we support solving sparse problems on GPU 😬 But at this point no need for using `_init_similar`, since we're not inferring type from the user input

Reply:
Similarly, `_init_similar(T[], ...)` should be replaced by `zero(T,...)`